### PR TITLE
(maint) cleanup rspec requirements

### PR DESCRIPTION
**Pull request by DavidS**
* Remove `gem` annotation from spec_helper.rb
* replace last usages of rspec2 collection matchers

### CODE REVIEW

**Comment by joshcooper** [comment]
Looks good though I'm curious how/why this was a problem? Is the rspec collections gem not compatible with module testing or something else?

**Comment by DavidS** [comment]
Updated the PR to highlight the two different changes, and add some rationale to the commits.

File: spec/lib/puppet_spec/matchers.rb [hunk 1]
```diff
@@ -1,18 +1,5 @@
 require 'stringio'
 
-########################################################################
-# Backward compatibility for Jenkins outdated environment.
-module RSpec
-  module Matchers
-    module BlockAliases
-      alias_method :to,     :should      unless method_defined? :to
-      alias_method :to_not, :should_not  unless method_defined? :to_not
-      alias_method :not_to, :should_not  unless method_defined? :not_to
```
Comment (joshcooper) [inline spec/lib/puppet_spec/matchers.rb]: This change seems unrelated to dropping rspec-collections?

Comment (joshcooper) [reply spec/lib/puppet_spec/matchers.rb]: I don't care so much, just glad to not be monkey patching rspec

Comment (DavidS) [reply spec/lib/puppet_spec/matchers.rb]: When I removed the collections, this flagged up as error and the tests are passing without so :arrow_right: :wastebasket:
